Makefile

```diff
@@ -1,20 +1,21 @@
 SHELL := /usr/bin/env bash -euo pipefail -c
 
-REPO_NAME    ?= $(shell basename "$(CURDIR)")
-PRODUCT_NAME ?= $(REPO_NAME)
+PRODUCT_NAME ?= consul-dataplane
 BIN_NAME     ?= $(PRODUCT_NAME)
 GOPATH       ?= $(shell go env GOPATH)
 GOBIN        ?= $(GOPATH)/bin
 
 # Get local ARCH; on Intel Mac, 'uname -m' returns x86_64 which we turn into amd64.
 # Not using 'go env GOOS/GOARCH' here so 'make docker' will work without local Go install.
 ARCH     ?= $(shell A=$$(uname -m); [ $$A = x86_64 ] && A=amd64; echo $$A)
-OS       ?= $(shell uname | tr [[:upper:]] [[:lower:]])
+# Only build for linux so that building the docker image works on M1 Macs.
+OS       ?= linux
```

Comment:
Is this effectively equivalent or is there anywhere else this got used other than the removed line below?

Reply:
Some things were failing to build for me on M1 because the OS does not match what we should build. We should always be building 'linux' as the OS. 

An example of [us forcing it in consul-k8s](https://github.com/hashicorp/consul-k8s/blob/main/Makefile#L44)